knoten/csm.py

```diff
@@ -139,6 +141,34 @@ def generate_latlon_boundary(camera, boundary, radii=None):
     alts : list
            List of altitude values
     '''
+    def generate_ground_point(image_pt, camera, dem, height, max_its, tolerance=0.0001):
+        # Assume some path to a dem has been provided
+        if not isinstance(image_pt, csmapi.ImageCoord):
+            # Support a call where image_pt is in the form (x,y)
+            image_pt = csmapi.ImageCoord(*image_pt)
+
+        if dem != '':
+            gd = GeoDataset(dem)
```

Comment:
It makes more sense to pass in the GeoDataset instead of the file name, dependency injection and all.

Reply:
`generate_latlon_boundary` can create the GeoDataset and then just pass it in each time. No need to re-create.